app/Models/Foundation/Summit/Registration/PromoCodes/Traits/SpeakersPromoCodeTrait.php

```diff
@@ -149,24 +144,64 @@ public function addUsage(string $owner_email, int $usage = 1)
      */
     public function removeUsage(int $to_restore, string $owner_email = null)
     {
-        try {
-            if ($owner_email == null)
-                throw new ValidationException("owner email is mandatory in order to remove usage for promo code {$this->getId()}");
+        if ($owner_email == null)
+            throw new ValidationException("owner email is mandatory in order to remove usage for promo code {$this->getId()}");
+
+        $existing_owner = $this->owners->filter(function ($e) use($owner_email){
+            return $e->getSpeaker()->getEmail() == $owner_email;
+        })->first();
+
+        if (!$existing_owner instanceof AssignedPromoCodeSpeaker)
+            throw new ValidationException("can't find an owner with the email {$owner_email} for the promo_code");
 
-            $existing_owner = $this->owners->filter(function ($e) use($owner_email){
-                return $e->getSpeaker()->getEmail() == $owner_email;
+        $existing_owner->clearRedeemedAt();
+
+        parent::removeUsage($to_restore);
+    }
+
+    /**
+     * @param string $email
+     * @param null|string $company
+     * @return bool
+     * @throws ValidationException
+     */
+    public function checkSubject(string $email, ?string $company):bool{
+        if($this->hasOwners()){
+            $existing_owner = $this->owners->filter(function ($e) use($email){
+                return $e->getSpeaker()->getEmail() == $email;
             })->first();
 
             if (!$existing_owner instanceof AssignedPromoCodeSpeaker)
-                throw new ValidationException("can't find an owner with the email {$owner_email} for the promo_code");
+                throw new ValidationException(sprintf('The Promo Code “%s” is not valid for the %s. Promo Code restrictions are associated with the purchaser email not the attendee.', $this->getCode(), $email));
+        }
+        return true;
```

Comment:
with this logic if does not has owner return true
and should return false

Reply:
done